nexus/db-model/src/update_artifact.rs

```diff
@@ -7,14 +7,16 @@ use crate::schema::update_available_artifact;
 use chrono::{DateTime, Utc};
 use omicron_common::api::internal;
 use parse_display::Display;
+use serde::Deserialize;
+use serde::Serialize;
 use std::io::Write;
 
 impl_enum_wrapper!(
     #[derive(SqlType, Debug, QueryId)]
     #[diesel(postgres_type(name = "update_artifact_kind"))]
     pub struct UpdateArtifactKindEnum;
 
-    #[derive(Clone, Copy, Debug, Display, AsExpression, FromSqlRow, PartialEq, Eq)]
+    #[derive(Clone, Copy, Debug, Display, AsExpression, FromSqlRow, PartialEq, Eq, Serialize, Deserialize)]
```

Comment:
Same here -- is this now more of a committed interface than it was?  Maybe not, since it seems like this was already in the database.

Reply:
I was relying on the determination of RFD 289 for this rationale too. Changing the structs within `db-model` would *already* be a breaking change - requiring the old version of Nexus to perform saga draining - so adding the serde derives here doesn't seem like a new constraint.